arcgis-ios-sdk-samples/Utility network/Display content of utility network container/DisplayContentUtilityNetworkContainerViewController.swift

```diff
@@ -0,0 +1,292 @@
+// Copyright 2021 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class DisplayContentUtilityNetworkContainerViewController: UIViewController, AGSGeoViewTouchDelegate, UIAdaptivePresentationControllerDelegate {
+    /// The map view managed by the view controller.
+    @IBOutlet var mapView: AGSMapView! {
+        didSet {
+            mapView.map = makeMap()
+        }
+    }
+    /// The bar button item to prompt return of the main view.
+    @IBOutlet var exitBarButtonItem: UIBarButtonItem!
+    @IBOutlet var legendBarButtonItem: UIBarButtonItem!
+    
+    /// A feature service for an electric utility network in Naperville, Illinois.
+    let featureServiceURL = URL(string: "https://sampleserver7.arcgisonline.com/arcgis/rest/services/UtilityNetwork/NapervilleElectric/FeatureServer")!
+    var utilityNetwork: AGSUtilityNetwork?
+    let graphicsOverlay = AGSGraphicsOverlay()
+    /// The default or previous viewpoint before entering the container view.
+    var previousViewpoint: AGSViewpoint?
+    /// An array containing information about the feature services' legends.
+    var legendInfos = [AGSLegendInfo]()
+    var featureLayers = [AGSFeatureLayer]()
+    
+    // The symbols used to display the container view contents.
+    let boundingBoxSymbol = AGSSimpleLineSymbol(style: .dash, color: .yellow, width: 3)
+    let attachmentSymbol = AGSSimpleLineSymbol(style: .dot, color: .green, width: 3)
+    let connectivitySymbol = AGSSimpleLineSymbol(style: .dot, color: .red, width: 3)
+    
+    /// The action that is prompted when exiting the container view.
+    @IBAction func exitContainerView() {
+        // Disable the bar button item since container view will be exited.
+        exitBarButtonItem.isEnabled = false
+        // Remove all the objects that were added onto the graphics overlay.
+        graphicsOverlay.graphics.removeAllObjects()
+        (mapView.map?.operationalLayers as? [AGSLayer])?.forEach { layer in
+            // Make each operational layer visible.
+            layer.isVisible = true
+            guard let previousViewpoint = previousViewpoint else { return }
+            // Return to the viewpoint before container view was entered.
+            mapView.setViewpointGeometry(previousViewpoint.targetGeometry) { [weak self] _ in
```

Comment:
1. As far as I know, yes, we can assume `previousViewpoint` is always non-nil when the exit container view button is enabled.
2. `setViewpoint` does not provide animation so I opted for `setViewpointGeometry`.
3. When the container view only has one element, `setViewpointCenter` is needed because `setViewpointGeometry` can't take care of point geometry without providing a scale.

Reply:
> `setViewpoint` does not provide animation so I opted for `setViewpointGeometry`.

Did you try using one of the `setViewpoint` methods that takes a completion handler? They should animate. If they don't, that's a bug.

Reply:
Oops, sorry please ignore my comment above. I was referring to the wrong section of code.